src/radical/pilot/pmgr/launching/default.py

```diff
@@ -888,42 +890,51 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand, tar_name):
             if virtenv_mode != 'private':
                 cleanup = cleanup.replace('v', '')
 
-        # use local VE ?
-        if virtenv_mode == 'local':
-            if os.environ.get('VIRTUAL_ENV'):
-                python_dist = 'default'
-                virtenv     = os.environ['VIRTUAL_ENV']
-            elif os.environ.get('CONDA_PREFIX'):
-                python_dist = 'anaconda'
-                virtenv     = os.environ['CONDA_PREFIX']
-            else:
-                # we can't use local
-                self._log.error('virtenv_mode is local, no local env found')
-                raise ValueError('no local env found')
+        # estimate requested resources
+
+        if os.environ.get('RADICAL_SAGA_SMT'):
```

Comment:
This could be simpler written as:
```py
system_architecture['smt'] = int(os.environ.get('RADICAL_SAGA_SMT', 1))
assert(system_architecture['smt'] > 0)
```
Exceptions can fall through IMHO

Reply:
kinda initial idea was that all possible exceptions will be handled here (key and value errors) Should we still assume that it could be non-convertible string variable, e.g., `None`? or just skip it and handle it as `int` only value?

p.s. also this part is how it was treated originally, thus we didn't force to set `system_architecture['smt']`, which was used at a certain cases. With the current setup it is required now. Will update it

Reply:
updated (added extra check, since if there is no env var, we need to check that there was value from config)